src/Orderings.jl

```diff
@@ -1,37 +1,29 @@
-module Orderings
-
 import Base.Order: lt, Ordering
-export LexicographicOrdering, LexicographicOrder, LenLexOrder
-
-abstract type LexicographicOrdering <: Ordering end
-
-
-## Assuming the order is given by dictionary with entries (::"letter", ::int)
-struct LexicographicOrder <: LexicographicOrdering
-    order::Dict
-end
+export LenLexOrder
 
+"""
+    struct LenLexOrder{T<:Alphabet} <: Ordering
 
-struct LenLexOrder{LexOrd<:LexicographicOrdering} <: Ordering
-    lexord::LexOrd
+Basic structure representing Length+Lexicographic (left-to-right) ordering of
+the words over given Alphabet. Lexicographing ordering of an Alphabet is
+implicitly specified inside Alphabet struct.
+"""
+struct LenLexOrder{T<:Alphabet} <: Ordering
+    A::T
 end
```

Comment:
Is there any reason to consider the subtypes of `Alphabet`?
```julia
struct LenLexOrder <: Ordering
    A::Alphabet
end
```

Reply:
`Alphabet` is an abstract type, `Alphabet{T}` is concrete; we want concrete types in structs.

I'd probably go with
```julia
struct LenLexOrder{T} <: Ordering
    A::Alphabet{T}
end
```